base_layer/core/src/transactions/transaction_components/unblinded_output.rs

```diff
@@ -216,30 +223,59 @@ impl UnblindedOutput {
         }
         let commitment = factories.commitment.commit(&self.spending_key, &self.value.into());
 
+        let range_proof = self.construct_range_proof(factories)?;
+
         let output = TransactionOutput::new(
             self.version,
             self.features.clone(),
             commitment,
-            RangeProof::from_bytes(
-                &factories
-                    .range_proof
-                    .construct_proof(&self.spending_key, self.value.into())?,
-            )
-            .map_err(|_| {
-                TransactionError::RangeProofError(RangeProofError::ProofConstructionError(
-                    "Creating transaction output".to_string(),
-                ))
-            })?,
+            range_proof,
             self.script.clone(),
             self.sender_offset_public_key.clone(),
             self.metadata_signature.clone(),
             self.covenant.clone(),
             self.encrypted_value.clone(),
+            self.minimum_value_promise,
         );
 
         Ok(output)
     }
 
+    fn construct_range_proof(&self, factories: &CryptoFactories) -> Result<RangeProof, TransactionError> {
```

Comment:
```suggestion
    fn construct_range_proof(&self, factories: &CryptoFactories, seed_nonce: Option<PrivateKey>) -> Result<RangeProof, TransactionError> {
```
These proofs must also contain the optional rewind the key